academy/xl-cosmjs/intro.md

```diff
@@ -0,0 +1,59 @@
+---
+title: "CosmJs Intro"
+order: 2
+description: What is CosmJs and what can it do for me?
+tag: intro
+---
+
+# The DApp stack
+
+Distributed applications, or DApps, are software applications that run on distributed networks. Blockchains provide persistent data, or state, as well as persistent processes and logic. Cosmos SDK helps developers create such applications. A user interface will be important in most cases. Server interactions are important in many cases. This is where [CosmJs](https://github.com/cosmos/cosmjs) can help.
```

Comment:
When are server interactions not important? Please elaborate.

Reply:
I'm not a huge fan of "unstoppable" logic because it seems a little hyperbolic and probably not true. I think "persistent" is a good adjective for the very reliable processes. It is a common adjective for stored data in and out of the blockchain space.

Data can be "transient" or "non-persistent" if you will lose it when you turn the computer off. Persistent data will still be there.

Reply:
Using the word twice isn't ideal. Open to ideas but I don't think it's far off as it is.

Reply:
"When are server interactions not important? Please elaborate."

This will devolve into a philosophical design principles argumentation because there are a lot of situations and a lot of apps. 

A server is NOT important if a DAPP maximalizes decentralization. A browser and a blockchain node is all you need. The client-side code (HTMNL etc) would be downloaded from IPFS, or similar. 

A server IS important if you want to support the front-end with analytics (GraphQL), cache the blockchain state (and analysis) for performance reasons, combine on-chain logic with off-chain sources (google maps, price APIs, order a pizza). While the blockchain can prove that you ordered a pizza, you probably need a server to actually contact the restaurant. 

In practice, most projects rely on server technology to deliver the UI. Unsatisfactory decentralization in some cases. Acceptable in others. 

You need CosmJS to build your UI or your JS-based server. 

Generally, I think it's clear to developers and I'm leery about rambling on about nuanced details that aren't vital for learning about CosmJs.